superset/superset/db_engine_specs.py

```diff
@@ -1151,7 +1151,12 @@ def fetch_data(cls, cursor, limit):
             raise Exception('Query error', state.errorMessage)
         try:
             return super(HiveEngineSpec, cls).fetch_data(cursor, limit)
-        except pyhive.exc.ProgrammingError:
+        except (pyhive.exc.ProgrammingError, pyhive.exc.OperationalError):
```

Comment:
I'm not sure it's a good idea. Let's check, in which cases `OperationalError` is returned. I believe that should be some cases when we should return it to a user/log instead of just skipping it.

Reply:
> Let's check, in which cases `OperationalError` is returned.

Actually, I didn't check the other cases when this error is raised, and I agree that in some cases we should return it.

My idea was this to be *temporary* until `gsc` handles this correctly without raising an exception on cancelation. I could have a look at the specific exception raised during cancelation and skip that one only. WDYT?

Reply:
If it's possible to catch only cancelation error it would be much better.

Reply:
Fixed [here](https://github.com/src-d/sourced-ui/pull/267/commits/eb80ddfdb9375a15af6fdae361ef0426c9f6e688).